### PR TITLE
Support changing fonts in the middle of a line. Use this to draw numeric

### PR DESCRIPTION
**Pull request by dabreegster**
IDs in the status bar as bold, not red.

Note IDs mostly show up in dev mode. Before:
Screenshot from 2021-02-24 21-06-44
After:
Screenshot from 2021-02-24 21-06-31

I got curious how vertical alignment works when you change fonts dramatically; looks like the SVG default is to align at the bottom, which feels reasonable. The purple background just shows we understand the max line height.
Screenshot from 2021-02-24 21-09-37

### CODE REVIEW

File: widgetry/src/text.rs [hunk 1]
```diff
@@ -441,32 +446,24 @@ impl Text {
...
-    .unwrap();
+    write!(&mut svg, r##"<text x="0" y="0" xml:space="preserve">"##,).unwrap();
 
     let mut contents = String::new();
     for span in spans {
         let fg_color = span.fg_color_for_style(&assets.style.borrow());
         write!(
             &mut contents,
-            r##"<tspan fill="{}" fill-opacity="{}" {}>{}</tspan>"##,
+            r##"<tspan font-size="{}" font-family="{}" {} fill="{}" fill-opacity="{}" {}>{}</tspan>"##,
```
Comment (dabreegster) [inline widgetry/src/text.rs:459]: This was... comically easy to support. I struggled a fair bit to get text rendering at all originally; it's pretty refreshing to come back to this code and fix something so easily.

File: game/src/common/mod.rs [hunk 2]
```diff
@@ -117,12 +117,11 @@ impl CommonState {
 
     fn osd_for(app: &App, id: ID) -> Text {
         let map = &app.primary.map;
-        let id_color = app.cs.bottom_bar_id;
         let mut osd = Text::new();
         match id {
             ID::Lane(l) => {
                 if app.opts.dev {
-                    osd.append(Line(l.to_string()).fg(id_color));
+                    osd.append(Line(l.to_string()).bold_body());
```
Comment (dabreegster) [inline game/src/common/mod.rs:124]: Maybe we should swap the bold/underline for IDs/names; I don't have a strong preference. But the red was pretty awful. :P

**Review by michaelkirk — APPROVED** [review]
The underline has kind of grown on me for the street names. It's reminiscent of marking the blank in a "fill in the blank" scenario, which is basically what we want to convey.

And arguably it makes sense that we'd treat dynamic, but not freeform, text fields like "Lane ID #3" differently, as you have here.

LGTM!

**Comment by michaelkirk** [comment]
> I got curious how vertical alignment works when you change fonts dramatically; looks like the SVG default is to align at the bottom, which feels reasonable. The purple background just shows we understand the max line height.

To be pedantic, it looks like a "baseline" alignment, which is slightly different from a "bottom" alignment. 🤓

**Pull request merged**